File: Java/src/main/java/OneLang/Parsers/Common/NodeManager/NodeManager.java
package OneLang.Parsers.Common.NodeManager;

import OneLang.Parsers.Common.Reader.Reader;

import java.util.List;
import OneLang.Parsers.Common.Reader.Reader;
import java.util.ArrayList;

public class NodeManager {
    public List<Object> nodes;
    public Reader reader;
    
    public NodeManager(Reader reader)
    {
        this.reader = reader;
        this.nodes = new ArrayList<Object>();
    }
    
    public void addNode(Object node, Integer start) {
        this.nodes.add(node);
    }
}

File: PHP/src/OneLang/Parsers/Common/NodeManager.php
<?php

namespace OneLang\Parsers\Common\NodeManager;

use OneLang\Parsers\Common\Reader\Reader;

class NodeManager {
    public $nodes;
    public $reader;
    
    function __construct($reader) {
        $this->reader = $reader;
        $this->nodes = array();
    }
    
    function addNode($node, $start) {
        $this->nodes[] = $node;
    }
}
